File: leetcode/2788.按分隔符拆分字符串.py
class Solution:
    def splitWordsBySeparator(self, words: List[str], separator: str) -> List[str]:
        ans = []
        for word in words:
            word = word.split(separator)
            ans += word
        ans1 = []
        for c in ans:
            if c != "":
                ans1.append(c)
        return ans1

File: leetcode/2788.按分隔符拆分字符串.cpp
class Solution {
public:
    vector<string> splitWordsBySeparator(vector<string>& words, char separator) {
        vector<string> ans;
        for (auto &w : words) {
            int j = 0;
            for(int i = 0; i <= w.size(); i++){
                if(i == w.size() || w[i] == separator){
                    if(i - j > 0) ans.push_back(w.substr(j, i - j));
                    j = i + 1;
                }
            }

        }
        return ans;
    }
};